### PR TITLE
Multitenant Gateway

### PR DESCRIPTION
**Pull request by kaloyan-raev**
These are the main changes in MinIO we need to run a multitenant S3 gateway:

1. Pass Access Key to the ObjectLayer through the Context https://github.com/minio/minio/pull/10615
1. Bypass Access Key validation
1. Bypass request signature validation

### CODE REVIEW

File: cmd/iam.go [hunk 1]
```diff
@@ -1125,7 +1125,7 @@ func (sys *IAMSys) SetUserSecretKey(accessKey string, secretKey string) error {
 func (sys *IAMSys) GetUser(accessKey string) (cred auth.Credentials, ok bool) {
 	objectAPI := newObjectLayerWithoutSafeModeFn()
 	if objectAPI == nil || sys == nil || sys.store == nil {
-		return cred, false
+		return cred, true
```
Comment (ifraixedes) [inline cmd/iam.go:1129]: Can we add a comment that clarifies why we need this at Storj?

File: cmd/signature-v4.go [hunk 2]
```diff
@@ -162,7 +162,7 @@ func doesPolicySignatureMatch(formValues http.Header) APIErrorCode {
 func compareSignatureV4(sig1, sig2 string) bool {
 	// The CTC using []byte(str) works because the hex encoding
 	// is unique for a sequence of bytes. See also compareSignatureV2.
-	return subtle.ConstantTimeCompare([]byte(sig1), []byte(sig2)) == 1
+	return true //subtle.ConstantTimeCompare([]byte(sig1), []byte(sig2)) == 1
```
Comment (ifraixedes) [inline cmd/signature-v4.go]: can we add a comment on the doc function that this function is disabled for fulfilling Storj requirements?

**Review by ifraixedes — COMMENTED** [review]
I've asked for some commends, however, it looks good to me.

File: cmd/iam.go (hunk 1 above)
Comment (kaloyan-raev) [reply cmd/iam.go:1129]: Comments added.

File: cmd/signature-v4.go (hunk 2 above)
Comment (kaloyan-raev) [reply cmd/signature-v4.go]: Comments added.

**Review by ifraixedes — APPROVED** [review]

**Review by mniewrzal — APPROVED** [review]

**Pull request merged**